File: 3633-earliest-finish-time-for-land-and-water-rides-i/3633. Earliest Finish Time for Land and Water Rides I.py
class Solution:
    def earliestFinishTime(self, landStartTime: List[int], landDuration: List[int], waterStartTime: List[int], waterDuration: List[int]) -> int:
        res = 1e9
        lst = sorted([(waterStartTime[i], waterDuration[i]) for i in range(len(waterStartTime))])
        waterStartTime = []
        waterDuration = []
        for x,y in lst:
            waterStartTime.append(x)
            waterDuration.append(y)
        print(waterStartTime)
        print(waterDuration)

        for i in range(len(landStartTime)):
            for j in range(len(waterStartTime)):
                if landStartTime[i] + landDuration[i] >= waterStartTime[j]:
                    res = min(res, landStartTime[i] + landDuration[i] + waterDuration[j])
                else:
                    res = min(res, waterStartTime[j] + waterDuration[j])

                if waterStartTime[j] + waterDuration[j] >= landStartTime[i]:
                    res = min(res, waterStartTime[j] + waterDuration[j] + landDuration[i])
                else:
                    res = min(res, landStartTime[i] + landDuration[i])
        return res

File: 3633-earliest-finish-time-for-land-and-water-rides-i/3633. Earliest Finish Time for Land and Water Rides I.java
class Solution {
    public int earliestFinishTime(int[] landStartTime, int[] landDuration, int[] waterStartTime, int[] waterDuration) {
        int res = Integer.MAX_VALUE;
        for (int i = 0; i < landStartTime.length; i++) {
            for (int j = 0; j < waterStartTime.length; j++) {
                if (landStartTime[i] + landDuration[i] >= waterStartTime[j]) {
                    res = Math.min(res, landStartTime[i] + landDuration[i] + waterDuration[j]);
                } else {
                    res = Math.min(res, waterStartTime[j] + waterDuration[j]);
                }
            }
        }

        for (int i = 0; i < waterStartTime.length; i++) {
            for (int j = 0; j < landStartTime.length; j++) {
                if (waterStartTime[i] + waterDuration[i] >= landStartTime[j]) {
                    res = Math.min(res, waterStartTime[i] + waterDuration[i] + landDuration[j]);
                } else {
                    res = Math.min(res, landStartTime[j] + landDuration[j]);
                }
            }
        }
        
        return res;
    }
}